pkg/dialer/factory.go

```diff
@@ -69,17 +69,17 @@ func IsCloudDriver(cluster *v3.Cluster) bool {
 }
 
 func IsPublicCloudDriver(cluster *v3.Cluster) bool {
-	return IsCloudDriver(cluster) && HasPublicAPIEndpoint(cluster)
+	return IsCloudDriver(cluster) && !HasOnlyPrivateAPIEndpoint(cluster)
 }
 
-func HasPublicAPIEndpoint(cluster *v3.Cluster) bool {
+func HasOnlyPrivateAPIEndpoint(cluster *v3.Cluster) bool {
 	switch cluster.Status.Driver {
 	case v32.ClusterDriverEKS:
-		return cluster.Status.EKSStatus.UpstreamSpec == nil ||
-			cluster.Status.EKSStatus.UpstreamSpec.PublicAccess == nil ||
-			*cluster.Status.EKSStatus.UpstreamSpec.PublicAccess
+		return cluster.Status.EKSStatus.UpstreamSpec != nil &&
+			cluster.Status.EKSStatus.UpstreamSpec.PublicAccess != nil &&
+			!*cluster.Status.EKSStatus.UpstreamSpec.PublicAccess
```

Comment:
I wonder if you should check private access too. I think public should be enough because if public is off then private should be on. Or roughly, public being off means it _is_ private.

Reply:
For EKS, you can't create a cluster with both private and public off.